qiling/os/windows/fncc.py

```diff
@@ -12,19 +12,92 @@
+               'PBOOL': 'POINTER', 'LPTOP_LEVEL_EXCEPTION_FILTER': 'DWORD', '_EXCEPTION_POINTERS': 'POINTER',
+               'ULONG_PTR': 'POINTER', 'PVECTORED_EXCEPTION_HANDLER': 'HANDLE', 'PFLS_CALLBACK_FUNCTION': 'POINTER',
+               'LPWIN32_FIND_DATAA': 'POINTER', 'LPOVERLAPPED': 'POINTER', 'LPCVOID': 'POINTER',
+               'HEAP_INFORMATION_CLASS': 'UINT', 'PSLIST_HEADER': 'POINTER', 'HMODULE': 'HANDLE', 'HRSRC': 'POINTER',
+               'HGLOBAL': 'POINTER', 'PMEMORY_BASIC_INFORMATION': 'POINTER', 'LPWCH': 'POINTER',
+               'LPSTARTUPINFOA': 'POINTER', 'LPSTARTUPINFOW': 'POINTER', 'LPTHREAD_START_ROUTINE': 'POINTER',
+                'LPCONTEXT': 'POINTER', 'LARGE_INTEGER': 'POINTER', 'LPMODULEINFO': 'POINTER', 'LPCWCH': 'POINTER',
+                'LPWORD': 'POINTER', 'LCID': 'POINTER', 'LPCCH': 'POINTER', 'LPBOOL': 'POINTER',
+                'LPCRITICAL_SECTION': 'POINTER', 'PSRWLOCK': 'POINTER', 'LPOSVERSIONINFOA': 'STRING',
+               'LPOSVERSIONINFOW': 'WSTRING', 'LPSYSTEM_INFO': 'POINTER', 'LPSYSTEMTIME': 'POINTER',
+                'LPPROCESSENTRY32W': 'POINTER', 'HLOCAL': 'POINTER', 'UINT_PTR': 'POINTER',
+               'LPOSVERSIONINFOEXW': 'POINTER', 'DWORDLONG': 'ULONGLONG', 'LPCPINFO': 'POINTER',
+                'LPNLSVERSIONINFO': 'POINTER', 'PCNZCH': 'STRING'}
+
+    if type in reptypedict.keys():
+        if type not in specialtype.keys():
+            return reptypedict[type]
+        else:
+            return specialtype[type]
+    else:
+        return type
+
 # x86/x8664 PE should share Windows APIs
-def winapi(cc, param_num=None, params=None):
+def winsdkapi(cc, param_num=None, dllname=None, specialtype=None, specialtypeEx=None, defparams=None):
     """
     @cc: windows api calling convention, only x86 needs this, x64 is always fastcall
-    @params: params dict
     @param_num: the number of function params, used by variadic functions, e.g printf
+    @dllname: the name of function
+    @funcname: function's name
+    @specialtype: customize replace type, e.g specialtype={'int':'UINT'} means repalce 'int' to 'UINT'
+    @specialtypeEx: customize replace param_name's type, e.g specialtypeEx={'time':'int'} means
+                replace the original type of time to int
+    @defparams: customize all params and their type
     """
-
     def decorator(func):
         @wraps(func)
         def wrapper(*args, **kwargs):
+            funcname = func.__name__[5:]
+            params = {}
+            paramlist = None
             ql = args[0]
-                                
+            if defparams is not None:
+                params = defparams
+            else:
+                if dllname is not None:
+                    root_path = os.path.abspath(os.path.join(os.getcwd(), "../.."))
+                    with open(root_path+'/qiling/extensions/windows_sdk/'+dllname+'.json', 'r') as f:
```

Comment:
use a universal method, check how gdb get the xml path. 

https://github.com/qilingframework/qiling/blob/cf8f5bfb8634c45fa6ee0fe8c609053b61fd0661/qiling/extensions/debugger/gdbserver/gdbserver.py#L405


also + dllname + , with a space